### ISSUE TITLE
Swift codegen errors on version 0.18.6

### ISSUE THREAD

**Issue by ethansinjin** [apollographql/apollo-codegen#383]
After upgrading to apollo 0.8.0 and apollo-codegen 0.18.6, my API.swift file isn't compiling correctly (Swift 4.0) due to the new (more explicit) `map` format.

Relevant Schema:

```
type Query {
pollTriggers: [PollTrigger!]
}

interface Node {
id: ID!
}

type PollTrigger implements Node {
id: ID!
poll: Poll!
filters: [PollTriggerFilters!]!
}

type PollTriggerFilters {
field: String
expectedValue: String
}

type Poll implements Node {
id: ID!
question: String!
choices: [String!]!
points: Int!
didParticipate: Boolean!
}
```

GraphQLQuery:
```
query PollTriggers {
    pollTriggers {
        id
        poll {
            ...pollInfo
        }
        filters {
            field
            expectedValue
        }
    }
}

fragment pollInfo on Poll {
    id
    question
    choices
    points
    didParticipate
}
```

Old API.swift:
```
public var pollTriggers: [PollTrigger]? {
      get {
        return (snapshot["pollTriggers"] as? [Snapshot]).flatMap { $0.map { PollTrigger(snapshot: $0) } }
      }
      set {
        snapshot.updateValue(newValue.flatMap { $0.map { $0.snapshot } }, forKey: "pollTriggers")
      }
    }
```

New API.swift:
```
public var pollTriggers: [PollTrigger]? {
      get {
        return (snapshot["pollTriggers"] as? [Snapshot]).flatMap { (value: [Snapshot]) -> [PollTrigger] in value.map { PollTrigger(snapshot: (value: Snapshot) -> PollTrigger **in** value) } }
      }
      set {
        snapshot.updateValue(newValue.flatMap { (value: [PollTrigger]) -> [Snapshot] in value.map { (value: PollTrigger) -> Snapshot in value.snapshot } }, forKey: "pollTriggers")
      }
    }
```

The relevant Swift errors:

```
Expected ',' separator
Expected expression in list of expressions
```

Both errors are occurring at the `**in**` in the getter for pollTriggers

**Comment by witekbobrowski** [comment]
I am having exact same issue.

**Comment by martijnwalraven** [comment]
This should be fixed on 0.18.7!

**Comment by matteinn** [comment]
thank you @martijnwalraven , this actually solved all of my issues
